### PR TITLE
only rewrite package.json if it has changed

### PR DESCRIPTION
**Pull request by tamird**
@guybedford

### CODE REVIEW

File: lib/config/package.js [hunk 1]
```diff
@@ -227,8 +227,15 @@ PackageJSON.prototype.write = function() {
     this.originalPjson.jspm = pjson;
   }
 
-  // NB check that the file hasn't changed since we opened it and if so, prompt
-  return asp(fs.writeFile)(this.fileName, stringify(this.originalPjson) + config.newLine);
+  asp(fs.readFile)(self.fileName).then(function(content) {
+    if (content === stringify(this.originalPjson) + config.newLine) {
```
Comment (guybedford) [inline lib/config/package.js]: `this` must be `self`.

Comment (guybedford) [inline lib/config/package.js]: This may also need to be `content.toString()` to do a correct equality.

File: lib/config/package.js [hunk 2]
```diff
@@ -227,8 +227,15 @@ PackageJSON.prototype.write = function() {
...
+      return Promise.resolve();
+    } else {
+      return asp(fs.writeFile)(this.fileName, stringify(this.originalPjson) + config.newLine);
+    }
+  }).catch(function() {
+    return asp(fs.writeFile)(this.fileName, stringify(this.originalPjson) + config.newLine);
```
Comment (guybedford) [inline lib/config/package.js]: What is this case for? We should specific the file code error specifically that you are trying to protect against here (if any?).

**Comment by tamird** [comment]
@guybedford updated, PTAL.

**Comment by guybedford** [comment]
:+1:

**Comment by tamird** [comment]
Any chance of getting this released?

**Pull request merged**